toolkit/tools/internal/file/file.go

```diff
@@ -89,6 +89,72 @@ func Copy(src, dst string) (err error) {
 	return NewFileCopyBuilder(src, dst).Run()
 }
 
+// CopyDir copies src directory to dst, creating the dst directory if needed.
+// dst is assumed to be a directory and not a file. Will preserve permissions.
+func CopyDir(src, dst string, dirPermissions, childPermissions fs.FileMode) (err error) {
+	isDstExist, err := PathExists(dst)
+	if err != nil {
+		return err
+	}
+	if isDstExist {
+		isDstDir, err := IsDir(dst)
+		if err != nil {
+			return err
+		}
+		if isDstDir {
+			logger.Log.Infof("destination (%s) already exists and is a directory", dst)
+			// Get dst dir info
+			dirInfo, err := os.Stat(dst)
+			if err != nil {
+				return fmt.Errorf("error getting dir info: %w", err)
+			}
+			// Set dirPermissions to the that of the exiting dst dir
+			dirPermissions = dirInfo.Mode().Perm()
+
+		}
+	}
+
+	if !isDstExist {
+		// Create dst dir
+		err = os.MkdirAll(dst, dirPermissions)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	// Open the source directory
+	entries, err := os.ReadDir(src)
+	if err != nil {
+		return err
+	}
+
+	// Iterate over the entries in the source directory
+	for _, entry := range entries {
+		srcPath := filepath.Join(src, entry.Name())
+		dstPath := filepath.Join(dst, entry.Name())
+
+		if entry.IsDir() {
+			// If it's a directory, recursively copy it
+			if err := CopyDir(srcPath, dstPath, childPermissions, childPermissions); err != nil {
+				return err
+			}
+		} else {
+			// If it's a file, copy it
+			if err := Copy(srcPath, dstPath); err != nil {
+				return err
+			}
+			// Set file permissions
+			if err := os.Chmod(dstPath, childPermissions); err != nil {
```

Comment:
You should use `NewFileCopyBuilder` to copy the file, so that you don't need to reimplement setting the file permissions.

Reply:
implemented with `NewFileCopyBuilder`